pkg/endpoint/bpf.go

```diff
@@ -974,37 +1064,20 @@ func (e *Endpoint) syncPolicyMap() error {
 			if err != nil {
 				e.getLogger().WithError(err).Errorf("Failed to delete PolicyMap key %s", entry.Key.String())
 				errors = append(errors, err)
-			} else {
```

Comment:
Why is this change necessary?

Reply:
Strictly not necessary, but delete is not needed when realized policy pointer is the same as the desired policy pointer. We just tested that the key is not present, so there is no point deleting it.

selector cache plumbing changed the policy lifecycle to copy the desired policy pointer to the realized policy pointer when the policy has been realized.